server/path_release.go

```diff
@@ -266,113 +270,6 @@ func getTrdlConfig(gitRepo *git.Repository, gitTag, trdlPath string) (*config.Tr
 	return cfg, nil
 }
 
-func buildReleaseArtifacts(ctx context.Context, tarWriter *nio.PipeWriter, gitRepo *git.Repository, fromImage string, runCommands []string, logger hclog.Logger) (error, func() error) {
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
-	if err != nil {
-		return fmt.Errorf("unable to create docker client: %w", err), nil
-	}
-
-	serviceDirInContext := ".trdl"
-	serviceDockerfilePathInContext := path.Join(serviceDirInContext, "Dockerfile")
-	serviceLabels := map[string]string{
-		"vault-trdl-release-uuid": uuid.NewV4().String(),
-	}
-
-	contextBuf := buffer.New(64 * 1024 * 1024)
-	contextReader, contextWriter := nio.Pipe(contextBuf)
-
-	go func() {
-		if err := func() error {
-			tw := tar.NewWriter(contextWriter)
-
-			logboek.Context(ctx).Default().LogF("Adding git worktree files to the build context\n")
-			logger.Debug("Adding git worktree files to the build context")
-
-			if err := trdlGit.AddWorktreeFilesToTar(tw, gitRepo); err != nil {
-				return fmt.Errorf("unable to add git worktree files to tar: %w", err)
-			}
-
-			dockerfileOpts := docker.DockerfileOpts{
-				WithArtifacts: true,
-				Labels:        serviceLabels,
-			}
-			if err := docker.GenerateAndAddDockerfileToTar(tw, serviceDockerfilePathInContext, fromImage, runCommands, dockerfileOpts); err != nil {
-				return fmt.Errorf("unable to add service dockerfile to tar: %w", err)
-			}
-
-			if err := tw.Close(); err != nil {
-				return fmt.Errorf("unable to close tar writer: %w", err)
-			}
-
-			return nil
-		}(); err != nil {
-			if closeErr := contextWriter.CloseWithError(err); closeErr != nil {
-				panic(closeErr)
-			}
-			return
-		}
-
-		if err := contextWriter.Close(); err != nil {
-			panic(err)
-		}
-	}()
-
-	logboek.Context(ctx).Default().LogF("Building docker image with artifacts\n")
-	logger.Debug("Building docker image with artifacts")
-
-	response, err := cli.ImageBuild(ctx, contextReader, types.ImageBuildOptions{
-		Dockerfile:  serviceDockerfilePathInContext,
-		PullParent:  true,
-		NoCache:     true,
-		Remove:      true,
-		ForceRemove: true,
```

Comment:
We lose this opportunity when using the CLI, but I am unsure if it is a significant issue.